sn_client/src/audit/spend_dag.rs

```diff
@@ -318,6 +353,23 @@ impl SpendDag {
         Ok(royalties)
     }
 
+    /// Remove all edges from a Node in the DAG
+    fn reset_edges(&mut self, node: NodeIndex) {
+        let incoming: Vec<_> = self
+            .dag
+            .edges_directed(node, petgraph::Direction::Incoming)
+            .map(|e| e.id())
+            .collect();
+        let outgoing: Vec<_> = self
+            .dag
+            .edges_directed(node, petgraph::Direction::Outgoing)
+            .map(|e| e.id())
+            .collect();
+        for edge in incoming.into_iter().chain(outgoing.into_iter()) {
```

Comment:
there is no incoming nor outgoing edges to the node, does it mean the node become `dangling` ?
then what does this mean to the DAG ?

Reply:
Indeed, but we add proper edges just after the call to this function, this is just a cleanup before write kind of thing